DRAFT: apt non-root configuration and automatic version check and configuration  for SONiC build

guillaumelambert opened the pull request:
@collivier @odd22 @ferrieux 

pushing this as a draft to have your feedbacks

guillaumelambert committed:
[build] Use apt-get source instead of dget in Makefiles

- create an a apt alternate configuration directory during 'make init'
- use 'apt-get source' instead of dget in makefiles
- adapt packages versions accordingly in rules/XX.mk files

SONiC images compilation is currently designed to be performed with a
non-root user under Ubuntu systems.
As a result, there is no guarantee that debian repos are declared in
/etc/apt/sources.list on the Ubuntu host system used.
Although for this reason 'apt-get source' can not be directly used
(out of the box) to that purpose, the build process recompiles packages
for various GNU+Linux Debian distributions.
And the packages sources used are retrieved from more or less official
repositories since the current solution in SONiC is to use dget and URLs
from various locations.
As discussed in the following URL
https://serverfault.com/questions/447457/use-apt-get-source-on-a-debian-repo-without-using-etc-apt-source-list
this is the simplest solution but it has several drawbacks from a
maintenance and security standpoint, for example :
* maintaining manually these URLS since they can be removed by external
  sites owners
https://github.com/sonic-net/sonic-buildimage/issues/13281
* potential loose authentication of packages sources
https://github.com/sonic-net/sonic-buildimage/pull/13302

A more complex solution is prosposed (on the same URL) and adress these
drawbacks by using an alternate apt configuration directory that is writable
by non-root user. It allows here to:
* identify packages sources with official Debian Keyring GPG signatures
* use 'apt-get source <pkgsrc_name>' to retrieve packages without maintaining
  several specific URLs
* in a next step use 'apt-cache showsrc <pkgsrc_name>' to retrieve a
  list of available versions to ckeck packages availability during the
  'make init' phase

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert committed:
[build] Add isc-dhcp automatic version check and configuration

mk files in folder rules/ are parsed several during the 'make configure'
phase.
Thus, it is preferable to check and optionnaly set the package versions
declared there before during the 'make init' phase.
This will prevent from 'apt-get source' calls failures.

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert commented:
@odd22 please can we sync the master branch so the PR real changes becomes more readable ?

odd22 commented:
> @odd22 please can we sync the master branch so the PR real changes becomes more readable ?

Yes. Done. For next one, perform a `git fetch upstream; git rebase upstream/master` on your branch after add the main sonic-buildimage repo in you local git repository as `upstream`

guillaumelambert committed:
[build] Add bash and debootstrap automatic version check and configuration

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

keboliu committed:
[Mellanox] Skip the leftover hardware reboot cause in case of last boot is warm/fast reboot (#13246)

- Why I did it
In case of warm/fast reboot, the hardware reboot cause will NOT be cleared because CPLD will not be touched in this flow. To not confuse the reboot cause determine logic, the leftover hardware reboot cause shall be skipped by the platform API, platform API will return the 'REBOOT_CAUSE_NON_HARDWARE' instead of the "hardware" reboot cause.

- How I did it
Check the proc cmdline to see whether the last reboot is a warm or fast reboot, if yes skip checking the leftover hardware reboot cause.

- How to verify it
a. Manual test:
    - Perform a power loss
    - Perform a warm/fast reboot
    - Check the reboot cause should be "warm-reboot" or "fast-reboot" instead of "power loss"
b. Run reboot cause related regression test.

Signed-off-by: Kebo Liu <kebol@nvidia.com>

xumia committed:
[Bug] Fix SONiC installation failure caused by pip/pip3 not found (#13284)

The main issue is the pip/pip3 command cannot be found when the package is being installed by apt-get.
When using the dpkg install, the searching path is PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
When using the apt-get install, the searching path is PATH=/usr/sbin:/usr/bin:/sbin:/bin
But the pip/pip3 default path is at /usr/local/bin, so dpkg works, but apt-get not work.

How I did it
Export the path /usr/local/bin for pip/pip3.
Make the deb packages can be installed by apt-get.

prsunny committed:
[Dash] Fix a typo (#13325)

Fix a typo in yang for Dash

shdasari committed:
SONiC YANG model for RADIUS. (#12749)

#### Why I did it
Added SONiC YANG model for RADIUS. 
Fixes https://github.com/sonic-net/sonic-buildimage/issues/12477 

#### How I did it
Added the RADIUS and RADIUS_SERVER tables for global and per RADIUS server configuration. RADIUS statistics reside in COUNTERS_DB and are not part of the configuration. These are not a part of this PR.

#### How to verify it
Compiled sonic_yang_mgmt-1.0-py3-none-any.whl.

#### Description for the changelog
SONiC YANG model for RADIUS.

dprital committed:
[submodule] Advance sonic-utilities pointer (#13333)

Update sonic-utilities submodule pointer to include the following:
* fb8f98b Preserve copp tables through DB migration ([#2524](https://github.com/sonic-net/sonic-utilities/pull/2524))
* 4aa512c [sfputil] Firmware download/upgrade CLI support for QSFP-DD (#1947) ([#2349](https://github.com/sonic-net/sonic-utilities/pull/2349))
* f63ef9a Revert sonic-utilities: Update config reload() to verify formatting of an input file (#2529) ([#2586](https://github.com/sonic-net/sonic-utilities/pull/2586))
* 3a09ecb [masic] 'show interfaces counters' reminds to use '-d all' option to check for internal links ([#2466](https://github.com/sonic-net/sonic-utilities/pull/2466))
* 65cf00a [storyteller] add link prober state change to story teller ([#2585](https://github.com/sonic-net/sonic-utilities/pull/2585))

Signed-off-by: dprital <drorp@nvidia.com>

guillaumelambert committed:
[build] Add kdump-tools and lm-sensors automatic version check and configuration

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert committed:
[build] Add snmpd automatic version check and configuration

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

pettershao-ragilenetworks committed:
[ragile] adapter for kernel 5.x (#10762)

Why I did it
Ragile adapter ra-b6510-32c ra-b6510-48v8c ra-b6910-64c ra-b6920-4s to kernel 5.x

Signed-off-by: “pettershao” pettershao@ragilenetworks.com

kenneth-arista committed:
[device/arista] Disabled polled_irq_mode for DNX SKUs (#13349)

Disabled polled_irq_mode for all Arista DNX devices as this mode
leads to excessive use of the CPU via an unneeded interrupt
polling thread.

qnos committed:
[Seastone] Enhancement fix for PR12200 syseeprom issue (#13344)

Why I did it
[Seastone] Enhancement fix for PR12200 syseeprom issue.

How I did it
Enhance the fix through replace the hardcoded devnum to bash variable

How to verify it
show platform syseeprom or decode-syseeprom

grahamhayes committed:
[Arista] Rely on automatic flash size detection for Raven (#13277)

Many of these switches have had flash upgraded beyond 2G however, in
boot0 both were assigned 2GB for legacy reasons.

Remove the hardcoding of the flash size and let boot0 autodetect the available space.

Signed-off-by: Graham Hayes <gr@ham.ie>

Signed-off-by: Graham Hayes <gr@ham.ie>

bingwang-ms committed:
[minigraph]: Support port name in ACL table AttachTo attribute (#13105)

Why I did it
This PR is to update minigraph.py to support both port alias and port name as input of AttachTo attribute of ACL table.
Before this change, only port alias is supported.

How I did it
Add a global variable to store port names
Search both port names and port alias wheh parsing the value of AttachTo.

How to verify it
Verified by a new unit test case test_minigraph_acl_attach_to_ports
Verified by copying the new minigraph.py to a testbed and run conflg load_minigraph.

xumia committed:
[Build] Support Debian snapshot mirror to improve build stability (#13097)

Why I did it
[Build] Support Debian snapshot mirror to improve build stability

It is to enhance the reproducible build, supports the Debian snapshot mirror. It guarantees all the docker images using the same Debian mirror snapshot and fixes the temporary build failure which is caused by remote Debain mirror indexes changed during the build. It is also to fix the version conflict issue caused by no fixed versions of some of the Debian packages.

How I did it
Add a new feature to support the Debian snapshot mirror.

How to verify it

guillaumelambert committed:
[build] Migrate iptables version download to apt-get source

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert committed:
[build] Migrate hiredis version download to apt-get source

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert committed:
[build] Migrate ifupdown2 version download to apt-get source

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert committed:
[build] Migrate iproute2 version download to apt-get source

Signed-off-by: Guillaume Lambert <guillaume.lambert@orange.com>

guillaumelambert commented:
superseded by 13381 on origin repo
